cirq-google/cirq_google/workflow/quantum_runtime.py

```diff
@@ -138,22 +178,63 @@ def __repr__(self) -> str:
         return _compat.dataclass_repr(self, namespace='cirq_google')
 
 
+def _safe_to_json(obj: Any, *, part_fn: str, nominal_fn: str, bak_fn: str):
+    """Safely update a json file.
+
+    1. The new value is written to a "part" file
+    2. The previous file atomically replaces the previous backup file, thereby becoming the
+       current backup file.
+    3. The part file is atomically renamed to the desired filename.
+    """
+    cirq.to_json_gzip(obj, part_fn)
+    if os.path.exists(nominal_fn):
+        os.replace(nominal_fn, bak_fn)
+    os.replace(part_fn, nominal_fn)
+
+
+def _update_updatable_files(
+    egr_record: ExecutableGroupResultFilesystemRecord,
+    shared_rt_info: SharedRuntimeInfo,
+    data_dir: str,
+):
+    """Safely update ExecutableGroupResultFilesystemRecord.json.gz and SharedRuntimeInfo.json.gz
+    during an execution run.
+    """
+    _safe_to_json(
+        shared_rt_info,
+        part_fn=f'{data_dir}/SharedRuntimeInfo.json.gz.part',
+        nominal_fn=f'{data_dir}/SharedRuntimeInfo.json.gz',
+        bak_fn=f'{data_dir}/SharedRuntimeInfo.json.gz.bak',
+    )
+    _safe_to_json(
+        egr_record,
+        part_fn=f'{data_dir}/ExecutableGroupResultFilesystemRecord.json.gz.part',
+        nominal_fn=f'{data_dir}/ExecutableGroupResultFilesystemRecord.json.gz',
+        bak_fn=f'{data_dir}/ExecutableGroupResultFilesystemRecord.json.gz.bak',
+    )
+
+
 def execute(
     rt_config: QuantumRuntimeConfiguration,
     executable_group: QuantumExecutableGroup,
     base_data_dir: str = ".",
 ) -> ExecutableGroupResult:
     """Execute a `cg.QuantumExecutableGroup` according to a `cg.QuantumRuntimeConfiguration`.
 
+    The ExecutableGroupResult's constituent parts will be persisted to disk as they become
```

Comment:
Nit: why not just "saved" instead of persisted ?

Reply:
done